konrad/cloud.py

```diff
@@ -472,79 +479,43 @@ def shift_cloud_profile(self, norm_new):
                 norm_new=norm_new,
             )
 
-    def update_cloud_profile(self, atmosphere, convection, **kwargs):
+    def update_cloud_profile(self, atmosphere, convection, radiation,
+                             **kwargs):
         """Keep the cloud profile fixed with model level (pressure). """
-        return
+        if self.coupling == 'convection':
+            self.shift_cloud_profile(
+                norm_new=convection.get('convective_top_index')[0]
+            )
+        elif self.coupling == 'temperature':
+            self.shift_cloud_profile(
+                norm_new=atmosphere.get_triple_point_index(),
+            )
+        elif self.coupling == 'subsidence':
+            Qr = radiation['net_htngrt_clr'][-1]
+            self.shift_cloud_profile(
+                norm_new=atmosphere.get_subsidence_convergence_max_index(Qr),
+            )
+        elif self.coupling == 'pressure':
+            return
 
 
 class HighCloud(DirectInputCloud):
     """Representation of a high-level cloud."""
-    def update_cloud_profile(self, atmosphere, convection, **kwargs):
-        """Keep the cloud attached to the convective top. """
-        self.shift_cloud_profile(
-            norm_new=convection.get('convective_top_index')[0]
-        )
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, coupling='subsidence', **kwargs)
```

Comment:
This may be the only part that you want to argue about 😉 
I recently changed back to defining the convective top based on the clear-sky divergence, as this is more robust when using MCICA. You can go back to your old `HighCloud` by passing `coupling="convection"` when creating the cloud object.

Reply:
Haha, yes I'd be happier if the convective top coupling was the default! Actually, I'd go as far as to say that it is actually the better choice for transient times, because using the clear sky subsidence as approximated from the radiative heating rates assumes energy balance of the free troposphere.
On the other hand, I can see that with McICA the convective top is not stable between iterations, so for you maybe it does make more sense to use the clear sky subsidence...